java/ql/src/experimental/Security/CWE/CWE-200/AndroidWebResourceResponse.qll

```diff
@@ -0,0 +1,82 @@
+/** Provides Android methods relating to web resource response. */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+
+/**
+ * The Android class `android.webkit.WebResourceRequest` for handling web requests.
+ */
+class WebResourceRequest extends RefType {
+  WebResourceRequest() { this.hasQualifiedName("android.webkit", "WebResourceRequest") }
+}
+
+/**
+ * The Android class `android.webkit.WebResourceResponse` for rendering web responses.
+ */
+class WebResourceResponse extends RefType {
+  WebResourceResponse() { this.hasQualifiedName("android.webkit", "WebResourceResponse") }
+}
+
+/** The `shouldInterceptRequest` method of Android's `WebViewClient` class. */
+class ShouldInterceptRequestMethod extends Method {
+  ShouldInterceptRequestMethod() {
+    this.hasName("shouldInterceptRequest") and
+    this.getDeclaringType().getASupertype*() instanceof TypeWebViewClient and
+    this.getReturnType() instanceof WebResourceResponse
```

Comment:
According to https://developer.android.com/reference/android/webkit/WebViewClient, both versions of `shouldInterceptRequest` return a `WebResourceResponse`, so this seems redundant.

```suggestion
    this.getDeclaringType().getASupertype*() instanceof TypeWebViewClient
```

Reply:
Changed